src/app.tsx

```diff
@@ -85,45 +86,117 @@ function Main({
           justifyContent: 'center',
         }}
       >
-        <ReferendumDeck
+        <ReferendaDeck
           network={network}
-          tracks={tracks}
           referenda={referenda}
+          tracks={tracks}
           voteOn={voteOn}
         />
       </div>
-      <ActionBar
-        left={referenda.size}
-        onAccept={() => voteOn(topReferenda, VoteType.Aye)}
-        onRefuse={() => voteOn(topReferenda, VoteType.Nay)}
-      />
-      <Spacer y={1} />
-    </Suspense>
+      {referenda.length > 0 && (
+        <ActionBar
+          left={referenda.length}
+          onAccept={() => topReferenda && voteOn(topReferenda, Vote.Aye)}
+          onRefuse={() => topReferenda && voteOn(topReferenda, Vote.Nay)}
+        />
+      )}
+    </>
   );
 }
 
+enum State {
+  LOADING,
+  STARTED,
+}
+
+type StartedContext = {
+  state: State.STARTED;
+  tracks: Map<number, Track>;
+  referenda: Map<number, ReferendumOngoing>;
+  votes: Map<number, Vote>;
+};
+
+export type Context = { state: State.LOADING } | StartedContext;
+
+function voteOn(
+  index: number,
+  vote: Vote,
+  setContext: React.Dispatch<React.SetStateAction<StartedContext>>
+) {
+  setContext((context) => {
+    context.votes.set(index, vote);
+    return { ...context }; // Copy object to trigger re-draw
+  });
+}
+
+function AppPanel({
+  context,
+  setContext,
+}: {
+  context: Context;
```

Comment:
Is the idea to have this as a react context or just is named as context. Any reason we pass this [context, setContext ] around, instead of just passing the voteOn or e.g. a voteHandler?

Reply:
Main Idea is mostly to have a store of static data fetch during the initial loading. The name is probably not the best :)

I will modify to only pass a handler.